From a1b2c3d4e5f60718293a4b5c6d7e8f9012345678 Mon Sep 17 00:00:00 2001
From: Tanya Singh <tanya_singh@accton.com>
Date: Thu, 23 Jul 2026 00:00:00 +0000
Subject: [PATCH] mediatek: add thermal protection init for EAP115/EAP115a

Add a boot-time init script that programs hwmon thermal trip points for
Edgecore EAP boards. On EAP115/EAP115a the 2.4G sensor trips at 105C and
restores at 95C.
---
 .../filogic/base-files/etc/init.d/thermal_protection | 14 ++++++++++++++
 1 file changed, 14 insertions(+)
 create mode 100755 target/linux/mediatek/filogic/base-files/etc/init.d/thermal_protection

diff --git a/target/linux/mediatek/filogic/base-files/etc/init.d/thermal_protection b/target/linux/mediatek/filogic/base-files/etc/init.d/thermal_protection
new file mode 100755
index 0000000000..84030929c6
--- /dev/null
+++ b/target/linux/mediatek/filogic/base-files/etc/init.d/thermal_protection
@@ -0,0 +1,14 @@
+#!/bin/sh /etc/rc.common
+
+START=99
+
+boot() {
+	case $(board_name) in
+	edgecore,eap115|\
+	edgecore,eap115a)
+		# Only set for 2.4G, trigger temp 105, restore temp 95
+		echo 95000 > /sys/class/hwmon/hwmon1/temp1_crit
+		echo 105000 > /sys/class/hwmon/hwmon1/temp1_max
+		;;
+	esac
+}
-- 
2.45.2
